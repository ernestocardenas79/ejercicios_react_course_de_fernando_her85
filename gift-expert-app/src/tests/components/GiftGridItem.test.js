import React from 'react';
import {shallow} from 'enzyme';
const { GiftGridItem } = require("../../components/GiftGridItem")

describe('Pruebas es Grift Grid Item', () => {

    const title='Un titulo';
    const url = 'https://algunaUr.com';
    const wrapper = shallow(<GiftGridItem title={title} url={url}/>);

    test('Deberia mostrar el componente corectamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de tener un parrafo con el title', ()=>{
        const p = wrapper.find('p');

        expect(p.text().trim()).toBe(title);
    })

    test('Debe obtner la imagen igual a la url y al tipo de los props ', () => {
            const img = wrapper.find('img');

            expect(img.prop('src')).toBe(url);
            expect(img.prop('alt')).toBe(title);
    })

    test('debe de tener animate_fadeIn ', () => {
        const div = wrapper.find('div');
        const classNames = div.prop('className');
        
        expect(classNames.includes('animate__fadeIn')).toBe(true);
    })    
})
