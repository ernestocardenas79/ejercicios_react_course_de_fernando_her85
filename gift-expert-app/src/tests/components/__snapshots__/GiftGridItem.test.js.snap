// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas es Grift Grid Item Deberia mostrar el componente corectamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://algunaUr.com"
  />
  <p>
    Un titulo
  </p>
</div>
`;
